maestro-orchestra/src/main/java/maestro/orchestra/yaml/YamlSetDarkMode.kt

```diff
@@ -0,0 +1,72 @@
+package maestro.orchestra.yaml
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.core.JsonParser
+import com.fasterxml.jackson.core.TreeNode
+import com.fasterxml.jackson.databind.DeserializationContext
+import com.fasterxml.jackson.databind.JsonDeserializer
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize
+import maestro.orchestra.DarkModeValue
+
+@JsonDeserialize(using = YamlSetDarkModeDeserializer::class)
+data class YamlSetDarkMode(
+    val value: DarkModeValue,
+    val label: String? = null,
+    val optional: Boolean = false,
+) {
+    companion object {
+        @JvmStatic
+        @JsonCreator(mode = JsonCreator.Mode.DELEGATING)
+        fun parse(value: DarkModeValue): YamlSetDarkMode {
+            return YamlSetDarkMode(value)
+        }
+    }
+}
+
+class YamlSetDarkModeDeserializer : JsonDeserializer<YamlSetDarkMode>() {
+
+    override fun deserialize(parser: JsonParser, ctxt: DeserializationContext): YamlSetDarkMode {
+        val mapper = (parser.codec as ObjectMapper)
+        val root: TreeNode = mapper.readTree(parser)
+        val input = root.fieldNames().asSequence().toList()
+        val label = getLabel(root)
+        when {
+            input.contains("value") -> {
+                val parsedValue = root.get("value").toString().replace("\"", "")
+                val returnValue = when (parsedValue) {
+                    "enabled" -> DarkModeValue.Enable
+                    "disabled" -> DarkModeValue.Disable
+                    else -> throwInvalidInputException(input)
+                }
+                return YamlSetDarkMode(returnValue, label)
+            }
+            (root.isValueNode && root.toString().contains("enabled")) -> {
+                return YamlSetDarkMode(DarkModeValue.Enable, label)
+            }
+            (root.isValueNode && root.toString().contains("disabled")) -> {
+                return YamlSetDarkMode(DarkModeValue.Disable, label)
+            }
+            else -> throwInvalidInputException(input)
+        }
+    }
+
+    private fun throwInvalidInputException(input: List<String>): Nothing {
+        throw IllegalArgumentException(
+            "setDarkMode command takes either: \n" +
+                    "\t1. enabled: To enable dark mode\n" +
+                    "\t2. disabled: To disable dark mode\n" +
+                    "\t3. value: To set dark mode to a specific value (enabled or disabled) \n" +
```

Comment:
I was a bit confused about this line

Reply:
I was copying the airplane mode function as it allows this syntax:

```yaml
- setDarkMode enabled
```

or

```yaml
- setDarkMode
    value: enabled
    label: "Something"
```

I'm definitely open to suggestions on how to improve this!

Reply:
Oh yeah I think the syntax makes sense. The 3rd line of the error message is what was confusing me - seems redundant?

```
1. enabled: To enable dark mode
2. disabled: To disable dark mode
3. value: To set dark mode to a specific value (enabled or disabled)
```

Reply:
Oh ok I think I understand now. Maybe it would be clearer to show those two styles of syntax as examples